generators/entity/index.js

```diff
@@ -494,7 +494,15 @@ class EntityGenerator extends BaseBlueprintGenerator {
     return {
       loadEntity() {
         // Update current context with config from file.
+        // copying everything will also overwrite search setting from the service itself
+        // which is wrong  for the microservice context.
+        this.context.disableElasticSearch =
+          !this.context.searchEngine && this.jhipsterConfig.applicationType === MICROSERVICE && this.context.entityExisted;
         Object.assign(this.context, this.entityStorage.getAll());
+        // in case search was deactiated on entity level (through  JSON file),   also disable  it in context
+        if (!this.entityConfig.searchEngine && this.context.entityExisted) {
+          this.context.searchEngine = false;
+        }
```

Comment:
Replace current changes with:
```
       loadEntity() {
-        // Update current context with config from file.
+        loadRequiredConfigIntoEntity(this.context, this.jhipsterConfig);
+
         Object.assign(this.context, this.entityStorage.getAll());
+
         this.loadDerivedAppConfig(this.context);
+
+        // Entity searchEngine support should opt-in instead of opt-out for gateway/microservice.
+        // The entity definition should take precedence so frontend and backend are in sync.
+        // Doesn't apply to microfrontends.
+        if (
+          this.entityConfig.searchEngine === undefined &&
+          !this.context.microfrontend &&
+          (this.context.applicationTypeMicroservice || (this.entityConfig.microserviceName && this.context.applicationTypeGateway))
+        ) {
+          this.info(`searchEngine is missing in .jhipster/${this.entityConfig.name}.json, should opt-in for gateway/microservice entities`);
+          this.context.searchEngine = false;
+        }
+
         this.loadDerivedClientConfig(this.context);
         this.loadDerivedServerConfig(this.context);
         this.loadDerivedPlatformConfig(this.context);
-        loadRequiredConfigIntoEntity(this.context, this.jhipsterConfig);
+
```

Reply:
those changes break one of unit tests:

```
  1) JHipster generator for entity
       regeneration from json file
         microservice
           with dto, service, pagination
             shall not generate search specific artifacts because elastic search is false on top level:

      AssertionError [ERR_ASSERTION]: src/main/java/com/mycompany/myapp/repository/search/FooSearchRepository.java exists
      + expected - actual
```


Because
`Object.assign(this.context, this.entityStorage.getAll());`

Copies all the values from entity to context and thus  activates search engine on the top level implicitely.   Line 499  was meant to prevent this and save state

So question is:  shall entity opt in activate elastic search if  service has deactivated it?

Reply:
Another issue,  check for this.context.entityExisted  is important to determine whether we are running interactively
( there are also unit tests for it ) - in this case entity shall take setting form the context